### PR TITLE
Core: Prevent saved stories from inheriting the original story's name

### LINKED ISSUE #35446: [Bug]: Creating a new story from args updates in the UI reuses the original story's `name` property
### Describe the bug

When creating a new story by updating args in the Storybook UI (Save from Controls → create new story), the newly created story reuses the `name` property from the original story. The two stories are then impossible to distinguish in the CSF file or in the sidebar.

This is a pre-existing edge case, not a 10.5 regression.

### Reproduction link

Observed on the baklava design-system repo during internal QA — no minimal reproduction.

### Reproduction steps

1. Open a story whose CSF export sets an explicit `name` (display name).
2. Change some args in the Controls panel and use Save from Controls to create a new story.
3. The generated export reuses the original story's `name`; the sidebar shows two entries with identical names.

### System

```bash
Storybook: 10.5.0-alpha.1 (pre-existing)
Project: baklava
Docgen: react-docgen-typescript (experimentalDocgenServer disabled)
```

### Additional context

Found during internal 10.5 QA. Suggested fix: detect the `name` property and update the save UI to expose separate inputs for the export name and the story display name.

### PR DESCRIPTION
**Pull request by thejamesgore**
Closes #35446

## What I did

When creating a new story via "Save from Controls", `duplicateStoryWithNewName`clones the original story's AST node and strips its `args` but not its `name`. 
If the original story has an explicit `name` the duplicate inherits it so the sidebar shows two stories with identical labels.

This PR removes the story objects own top level `name` property from the clone

The new story's display name then derives from its new export name which matches the features intent the save dialogs input is an export name, the response's `newStoryName`/`newStoryId` are already derived from it server side and the new story file generator emits no `name` property. 

Previously the file contradicted this and the response reported a display name derived from the new export name, while the story on disk kept the original's `name` and rendered under the old label. Now they match.

Covered shapes are the CSF3 object stories including `satisfies Story` / `as Story` wrappers and CSF4 `meta.story({...})` with both identifier (`name:`) and string literal (`'name':`) keys removed

Nested `name` keys such as `parameters.*.name` or `argTypes` entry for a prop called `name` are deliberately left untouched which is why this doesn't reuse the existing any-depth traversal that removes `args`.

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [x] unit tests
- [ ] integration tests
- [ ] end-to-end tests

Two new cases in `duplicate-story-with-new-name.test.ts` with dedicated mocks (`name-variances.stories.tsx`, `csf4-name-variances.stories.tsx`). They check that duplicates of named stories no longer inheriting `name` and nested `name` keys in `parameters`/`argTypes` are preserved. 
Both tests fail without the fix and the four pre existing test cases are unchanged.

#### Manual testing

1. Run a sandbox: `yarn task sandbox --template react-vite/default-ts --start-from auto`
2. Give a story an explicit name, e.g. `export const Primary = { name: 'Primary!', args: { ... } }`
3. Open it, change any control, and use "Create new story" with a new export name
4. Before: the sidebar shows two entries both labeled "Primary!". After: the
   new story appears under its own name derived from the export name you typed.

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

No docs change needed as this makes save from controls behave as documented and nothing is deprecated or removed.

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Declare whether manual QA will be needed for this PR during the next release, through `qa:needed` or `qa:skip`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/publish.yml) or locally with `gh workflow run --repo storybookjs/storybook publish.yml --field pr=<PR_NUMBER>`_

## Summary by CodeRabbit

* **Bug Fixes**
  * Duplicated stories no longer inherit the original top-level display name.
  * Nested `name` values in story parameters and argument types remain unchanged.
  * Improved duplication behavior across CSF/CSF4 formats, including stories using TypeScript wrappers.

* **Tests**
  * Added automated coverage for name-variation scenarios (CSF and CSF4) and edge cases like string-literal vs computed name keys.

### CODE REVIEW

File: code/core/src/core-server/utils/save-story/duplicate-story-with-new-name.ts [hunk 1]
```diff
@@ -49,6 +65,20 @@ export const duplicateStoryWithNewName = (csfFile: In, storyName: string, newSto
...
+  // display name. Nested `name` keys (parameters, argTypes) must be left untouched.
+  const storyObject = getStoryObjectExpression(cloned.init, isCsf4Story);
+  if (storyObject) {
+    storyObject.properties = storyObject.properties.filter(
+      (prop) =>
+        !(
+          t.isObjectProperty(prop) &&
+          ((t.isIdentifier(prop.key) && prop.key.name === 'name') ||
+            (t.isStringLiteral(prop.key) && prop.key.value === 'name'))
+        )
```
[the remark this answers is not in the document]
Comment (thejamesgore) [reply code/core/src/core-server/utils/save-story/duplicate-story-with-new-name.ts:78]: Done! I have added `!prop.computed` on the identifier branch as suggested, so dynamic `[name]` keys are preserved. Left the string-literal branch unguarded on purpose as `['name']` is still statically the name key, so it should be stripped. Both cases are covered in the new test.

File: code/core/src/core-server/utils/save-story/mocks/name-variances.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  argTypes: {
+    name: {
+      control: 'text',
+    },
+  },
+} satisfies Story;
+
+export const WithOnlyName = {
+  name: 'Only A Name',
+} satisfies Story;
```
[the remark this answers is not in the document]
Comment (thejamesgore) [reply code/core/src/core-server/utils/save-story/mocks/name-variances.stories.tsx:40]: Can't do this in the fixture file as oxfmt normalises quoted object keys (`'name':` → `name:`) on commit, which is why that fixture uses an identifier key. Added an inline-source test with `loadCsf` instead, covering the quoted key (removed), computed `['name']` (removed), and dynamic `[name]` (preserved).

**Comment by thejamesgore** [comment]
Danger is failing because I can't set labels myself and Danger wants `ci:*`/`qa:*` on this.

Could a maintainer add them please?
